### DISCUSSION TITLE
Taproot support in Dogecoin

### DISCUSSION THREAD

**Discussion by susruth** [dogecoin/dogecoin#3684] [General]
Hi everyone, there is a lot happening in the Bitcoin ecosystem leveraging taproot, it has been a decent while since it is added to Bitcoin as well. Is there any timeline we are looking at to support Taproot on Dogecoin or are there any major blockers that are still preventing us from supporting it?

**Comment by patricklodder** [comment]
There's never a timeline. However, I know of multiple people that are working in the background to propose enhancements that could ultimately enable taproot.

What do you want to do with taproot?

**Reply by susruth** [reply]
It would be great to have schnorr support to create decentralised bridges to other ecosystems using FROST, also it would be interesting to play around with covenants and other more complex primitives powering projects like BitVM, Babylon, etc.

**Comment by sandakersmann** [comment]
Jaqen Hash’ghar did warn us about SegWit in his amazing article back in 2016. Taproot and SegWit enabled 60% of the block to be open for competition just between JPEGs and other random data.

https://medium.com/the-publius-letters/segregated-witness-a-fork-too-far-87d6e57a4179

![JaqenAboutSegWitIn2016](https://github.com/user-attachments/assets/e76b784c-0044-40df-a682-1a6af050c90a)

**Reply by patricklodder** [reply]
I think "jpegs" is moot. Dogecoin doesn't have segwit or taproot yet a higher % of the tx volume is "jpegs" than on bitcoin. So you're right, but it doesn't matter. Not implementing segwit does not prevent "jpegs".

However, please read #2264, especially around this: https://github.com/dogecoin/dogecoin/discussions/2264#discussioncomment-831006. Most people there and that I've talked to outside of this repo can agree with an outcome where we DO segregate the signatures off the tx body (to prevent multi-party malleability) but DO NOT provide any economic incentives or extra block size.

Note that doginals has found a way to cross-link tx data in their protocol without even needing to put out a new policy, so whatever limit is imposed is ultimately arbitrary - all a tighter limit does is ultimately cause more bytes of spam overhead (and yes, with that cost, but we're talking sub-cent per kB). As long as there is no economic incentive (as in bytes==vBytes and max_vBlockSize==max_blocksize) then the economic downside described above be mitigated? 

Let me know what you think.

**Reply by sandakersmann** [reply]
The main issue is that random data does not have to compete with monetary transactions in BTC. This makes it a lot cheaper to embed data than doing normal transactions per kB. Multi-party malleability should be solved with Schnorr signatures like in BCH:

https://gist.github.com/markblundeberg/a3aba3c9d610e59c3c49199f697bc38b

**Reply by patricklodder** [reply]
So you're agreeing that the proposed mitigation would solve the main issue?

**Reply by sandakersmann** [reply]
Yes, but it's still a ugly hack.
